docs/sdk/src/reference_docs/tokens_in_substrate.rs

```diff
@@ -0,0 +1,97 @@
+//! - Where to go next
+//!
+//! ## Traits and Trait Implementations
+//!
+//! Broadly speaking, token logic in Substrate can be divided into two categories: traits and
+//! trait implementations.
+//!
+//! **Traits** define common interfaces that types of tokens should implement. For example, the
+//! [`frame_support::traits::fungible::Inspect`] trait specifies an interface for *inspecting*
+//! token state such as the total issuance of the token, the balance of individual accounts, etc.
+//!
+//! **Trait implementations** are concrete implementations of these traits. For example, one of the
+//! many traits [`pallet_balances`] implements is [`frame_support::traits::fungible::Inspect`].
+//!
+//! The distinction between traits and trait implementations is helpful because it allows pallets
+//! and other logic to be generic over their dependencies, avoiding cumbersome pallet tight
+//! coupling.
+//!
+//! To illustrate this with an example let's consider [`pallet_preimage`]. This pallet takes a
+//! deposit in exchange for storing some preimage for use later. A naive implementation of the
+//! pallet may use [`pallet_balances`] as a dependency, and directly call the methods exposed by
+//! [`pallet_balances`] to reserve and unreserve deposits. This approach works well, until someone
+//! has a use case requiring that an asset from a different pallet such as [`pallet_assets`] is
+//! used for the deposit. Rather than tightly couple [`pallet_preimage`] to [`pallet_balances`] AND
+//! [`pallet_assets`], along with every other token-handling pallet a user could possibly specify,
+//! [`pallet_preimage`] does not specify a concrete pallet as a dependency but instead accepts any
+//! dependency which implements the [`frame_support::traits::tokens::currency::ReservableCurrency`]
+//! trait. This allows [`pallet_preimage`] to support any arbitrary pallet implementing this trait,
+//! without needing any knowledge of what those pallets may be or requiring changes to support new
+//! pallets which may be written in the future.
+//!
+//! ## Fungible Token Traits in Substrate
+//!
+//! The [`frame_support::traits::fungible`] crate contains the latest set of Substrate
+//! fungible token traits, and is recommended to use for all new logic requiring a fungible token.
+//! See the crate documentation for more info about these fungible traits.
+//!
+//! [`frame_support::traits::fungibles`] provides very similar functionality to
+//! [`frame_support::traits::fungible`], except it supports managing multiple tokens.
+//!
+//! You may notice the trait [`frame_support::traits::Currency`] with similar functionality is also
+//! used in the codebase, however this trait is deprecated and existing logic is in the process of
+//! being migrated to [`frame_support::traits::fungible`] ([tracking issue](https://github.com/paritytech/polkadot-sdk/issues/226)).
+//!
+//! ## Fungible Token Trait Implementations in Substrate
```

Comment:
I would add a section here as "Rust Docs Advance Tip" that elaborates on the fact that the knowledge of "`pallet-balances` implements `funguble::Inspect`" is not some complicated knowledge that you have to know by heart or memorize in any way; One can simply look at the list of the implementors of _any_ trait and find the answer: 

https://paritytech.github.io/polkadot-sdk/master/frame/traits/tokens/fungible/trait.Mutate.html#implementors